Tests/NIOHTTP1Tests/HTTPServerUpgradeTests.swift

```diff
@@ -424,39 +425,50 @@ private class UpgradeResponseDelayer: HTTPServerProtocolUpgrader {
     }
 }
 
-private class UserEventSaver<EventType>: ChannelInboundHandler {
+private final class UserEventSaver<EventType: Sendable>: ChannelInboundHandler, Sendable {
     public typealias InboundIn = Any
-    public var events: [EventType] = []
+
+    private let _events = NIOLockedValueBox<[EventType]>([])
+
+    var events: [EventType] {
+        self._events.withLockedValue { $0 }
+    }
 
     public func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
-        events.append(event as! EventType)
+        self._events.withLockedValue { $0.append(event as! EventType) }
         context.fireUserInboundEventTriggered(event)
     }
 }
 
-private class ErrorSaver: ChannelInboundHandler {
-    public typealias InboundIn = Any
-    public typealias InboundOut = Any
-    public var errors: [Error] = []
+private final class ErrorSaver: ChannelInboundHandler, Sendable {
+    typealias InboundIn = Any
+    typealias InboundOut = Any
 
-    public func errorCaught(context: ChannelHandlerContext, error: Error) {
-        errors.append(error)
+    private let _errors = NIOLockedValueBox<[any Error]>([])
+
+    var errors: [Error] {
+        self._errors.withLockedValue { $0 }
+    }
+
+    func errorCaught(context: ChannelHandlerContext, error: Error) {
+        self._errors.withLockedValue { $0.append(error) }
         context.fireErrorCaught(error)
     }
 }
 
-private class DataRecorder<T>: ChannelInboundHandler {
+private final class DataRecorder<T: Sendable>: ChannelInboundHandler, Sendable {
     public typealias InboundIn = T
-    private var data: [T] = []
+
+    private let data = NIOLockedValueBox<[T]>([])
 
     public func channelRead(context: ChannelHandlerContext, data: NIOAny) {
```

Comment:
More access level weirdness here with the type being private but the methods being public.